### DISCUSSION TITLE
.merge_models requires pre-trained models to have the same embedding model?

### DISCUSSION THREAD

**Discussion by themeaningofmeaning** [MaartenGr/BERTopic#1727] [Q&A] [answered]
`merged_model = BERTopic.merge_models([bertopic_model_wiki, bertopic_model_arxiv])
`

When attempting to merge two pre-trained models (MaartenGr/BERTopic_Wikipedia and MaartenGr/BERTopic_ArXiv), I get an error, ` incompatible dimension for X and Y matrices: X.shape[1] == 768 while Y.shape[1] == 384`," which suggests a mismatch in the dimensions of the embeddings used by these two BERTopic models.  Digging in deeper, they both appear to use different embedding models (Cohere multilingual embedding model and a sentence-transformers/all-mpnet-base-v2 model), which would result in different dimensionalities of their respective embeddings.

However, when initializing each BERTopic model with a sentence transformer model that produces embeddings of the same size per the documentation, I'm still run into the same error:
```
# Define a common sentence transformer
sentence_model = SentenceTransformer('sentence-transformers/all-MiniLM-L6-v2')

# Load the first BERTopic model with the defined sentence transformer
bertopic_model_wiki = BERTopic.load("MaartenGr/BERTopic_Wikipedia", embedding_model=sentence_model)

# Load the second BERTopic model with the same sentence transformer
bertopic_model_arxiv = BERTopic.load("MaartenGr/BERTopic_ArXiv", embedding_model=sentence_model)

#merge the models (this still produces the same error around incompatible dimensions
merged_model = BERTopic.merge_models([bertopic_model_wiki, bertopic_model_arxiv])
```

Has anyone figured out a workaround other than retraining the models?

**Comment by MaartenGr** [comment, marked as answer]
This is what is currently expected of the `.merge_models` functionality. Embeddings are created based on the average embeddings of a cluster which cannot be easily generalized to another embedding model.
